### PR TITLE
Fix the bugs in list setters to clear then in case of empty input

### PR DESCRIPTION
**Pull request by suresh-prakash**
## Description
Currently, there is no ability to make the list of selections, sorts, from type expressions and aggregations empty through TransformedQueryBuilder. Fixed this bug.

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: document-store/src/main/java/org/hypertrace/core/documentstore/query/Query.java [hunk 1]
```diff
@@ -125,7 +125,9 @@ public QueryBuilder setSelection(final Selection selection) {
     }
 
     public QueryBuilder setSelections(final List<SelectionSpec> selectionSpecs) {
-      if (CollectionUtils.isNotEmpty(selectionSpecs)) {
+      if (CollectionUtils.isEmpty(selectionSpecs)) {
```
Comment (kotharironak) [inline document-store/src/main/java/org/hypertrace/core/documentstore/query/Query.java:128]: Was it crashing? what was the issue?

Comment (suresh-prakash) [reply document-store/src/main/java/org/hypertrace/core/documentstore/query/Query.java:128]: Basically, when we try to set an `emptyList()` as selection list, it was not being set properly.

File: document-store/src/test/java/org/hypertrace/core/documentstore/query/QueryBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  @Test
+  void testSetAndClearSelections() {
+    final Query queryWithSelections =
+        queryBuilder.addSelection(ConstantExpression.of("Something")).build();
+    assertEquals(1, queryWithSelections.getSelections().size());
+
+    final Query queryWithoutSelections = queryBuilder.setSelections(emptyList()).build();
+    assertEquals(0, queryWithoutSelections.getSelections().size());
```
Comment (kotharironak) [inline document-store/src/test/java/org/hypertrace/core/documentstore/query/QueryBuilderTest.java:33]: So, previously, this was failing, right?

**Review by kotharironak — APPROVED** [review]

File: document-store/src/test/java/org/hypertrace/core/documentstore/query/QueryBuilderTest.java (hunk 2 above)
Comment (suresh-prakash) [reply document-store/src/test/java/org/hypertrace/core/documentstore/query/QueryBuilderTest.java:33]: That's correct.

**Pull request merged**